### PR TITLE
Book index sorting

### PR DESCRIPTION
**Pull request by plapicola**
Adds sorting functionality for books index page. Completes #23 until reviews completed.

### CODE REVIEW

**Comment by plapicola** [comment]
@iandouglas @memcmahon
This appears to correctly be sorting the book index page, but the logic in the controller seems like it would be a very poor practice. Would you have any suggestions on how to make this cleaner?

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,7 +1,22 @@
 class BooksController < ApplicationController
 
   def index
-    @books = Book.all
+    case params[:sort]
+      when 'page_count'
+        if params[:order] == "desc"
```
Comment (iandouglas) [inline app/controllers/books_controller.rb]: what if you passed `params[:sort]` as an argument to `Book.by_page_count`

File: app/models/book.rb [hunk 2]
```diff
@@ -7,4 +7,21 @@ class Book < ApplicationRecord
   has_many :book_authors
   has_many :reviews
   has_many :authors, through: :book_authors
+
+  def self.by_page_count_ascending
```
Comment (iandouglas) [inline app/models/book.rb]: for that matter, you could DRY up these four methods by passing which field you want to sort by, plus the direction of the sort.

File: app/views/books/index.html.erb [hunk 3]
```diff
@@ -2,6 +2,8 @@
   <ul>
     <li><a href="<%= books_path %>?sort=page_count&order=asc">Sort by Page Count (Ascending)</a></li>
     <li><a href="<%= books_path %>?sort=page_count&order=desc">Sort by Page Count (Descending)</a></li>
+    <li><a href="<%= books_path %>?sort=year&order=asc">Sort by Year (Ascending)</a></li>
+    <li><a href="<%= books_path %>?sort=year&order=desc">Sort by Year (Descending)</a></li>
```
Comment (iandouglas) [inline app/views/books/index.html.erb]: try this:
```ruby
<li><%= link_to 'Sort by Year Descending', books_path, params: {sort: 'year', order: 'desc'} %></li>
```

**Comment by plapicola** [comment]
Added suggestions from @iandouglas for cleaning up the code. Thank you Ian for the quick feedback.

**Comment by juliamarco** [comment]
Nice job!

**Pull request merged**